app/components/autocomplete/at_mention/index.js

```diff
@@ -29,16 +51,34 @@ function mapStateToProps(state, ownProps) {
         }
     }
 
+    const match = postDraft.substring(0, cursorPosition).match(regex);
+    let matchTerm = null;
+    if (match) {
+        matchTerm = isSearch ? match[1] : match[2];
+    }
+
+    const opts = {
+        matchTerm,
+        currentUserId,
+        isSearch
+    };
+
+    if (isSearch) {
+        filterMembersInCurrentTeam(state, {...opts, array: teamMembers});
```

Comment:
These aren't memoizing anything because the second argument is changing constantly so they're recomputing these lists on every store change.

This is also not good since selectors are supposed to be effectively pure functions (pure other than the memoization) and I assume we're mutating the array here

Reply:
Okay update now that I've looked at the selectors themselves since they are salvageable. What you can do is:
1. Instead of passing in opts, just pass in the matchTerm as the second argument. The selectors can get currentUserId themselves if needed, and they don't use isSearch. That'll stop all the re-rendering since it'll only re-render if the state changes meaningfully or the term changes
2. Just have them return the array instead of passing it in and mutating it. That should be easy since thankfully we're not doing anything with the previous results; we're just emptying the array and refilling it  with new data currently.